catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraItems.scala

```diff
@@ -18,52 +18,60 @@ import scala.util.{Failure, Success}
 
 trait SierraItems extends Logging with SierraLocation {
   def extractItemData(
-    sierraTransformable: SierraTransformable): List[SierraItemData] = {
-    sierraTransformable.itemData.values
-      .map { _.data }
-      .map { jsonString =>
-        fromJson[SierraItemData](jsonString) match {
-          case Success(d) => Some(d)
-          case Failure(e) => {
-            error(s"Failed to parse item!", e)
-            None
+    sierraTransformable: SierraTransformable): Map[String, SierraItemData] =
+    sierraTransformable.itemData
+      .map { case (id, itemRecord) => (id, itemRecord.data) }
+      .map {
+        case (id, jsonString) =>
+          fromJson[SierraItemData](jsonString) match {
+            case Success(data) => Some(id -> data)
+            case Failure(e) => {
+              error(s"Failed to parse item!", e)
+              None
```

Comment:
It’s a discussion for another time, but I don't think we should silently drop malformed records here. We should made a dashboard go red, because this should really never happen.

Reply:
eek, straight to DLQ 🗑